test/plugin/yjsEditor.test.ts

```diff
@@ -0,0 +1,74 @@
+import { Node, Element } from 'slate';
+import { toSyncElement, YjsEditor } from '../../src';
+import { createTestEditor } from '../utils';
+
+// onChange fires in the next tick, see slate createEditor
+async function waitForSlateOnChangeAround(callback: () => void) {
+  await callback();
+  await Promise.resolve();
+}
+
+describe('YjsEditor', () => {
+  let yjsEditor: YjsEditor;
+
+  const initialValue: Node[] = [
+    { type: 'paragraph', children: [{ text: '' }] },
+  ];
+
+  const newElement: Element = {
+    type: 'paragraph',
+    children: [{ text: 'new' }],
+  };
+
+  beforeEach(async () => {
+    yjsEditor = await createTestEditor(initialValue);
+  });
+
+  it('should sync changes to the shared type with slate', async () => {
+    await waitForSlateOnChangeAround(() =>
+      yjsEditor.sharedType.push([toSyncElement(newElement)])
+    );
+
+    expect(yjsEditor.children).toEqual<Node[]>([...initialValue, newElement]);
+    // make sure we didn't erroneously add elements to the shared type - can happen
+    // if the observer we have for it applies "local" events
+    expect(yjsEditor.sharedType.toJSON()).toEqual<Node[]>(yjsEditor.children);
```

Comment:
Please use the helper method for converting a sharedType to a slate doc (this is essentially the same for now but I'm currently working on a change that will change that)